File: Java/VersionOneSSO/src/com/versionone/sso/IResponseParser.java
package com.versionone.sso;

import java.io.InputStream;

/**
 * Interface that a SSO provider (IDP and SP) needs to implement
 * @author jerry
 *
 */
public interface IResponseParser {

	/**
	 * 	The the Post URL from the form on the page
	 * @return
	 */
	String getPostUrl();

	/**
	 * This is the UriPartial.Authority part of the site that sent us the SAML response. 
	 * This is important when the PostsUrl is relative.
	 * @param value
	 */
	void setUrlAuthority(String value);

	/**
	 * CLoad the response received from a server
	 * @param reader 
	 * @throws Exception
	 */
	void loadResponse(InputStream reader) throws Exception;
	
	/**
	 * Return the form data necessary to take the next step.
	 */
	PostData getPostData();
}

File: Net/SSOExtension/VersionOne.SDK.APIClient.SSOExtension/IResponseParser.cs
using System.IO;

namespace VersionOne.SDK.APIClient.SSOExtension
{
    /// <summary>
    /// Interface that a SSO provider (IDP and SP) needs to implement
    /// </summary>
    public interface IResponseParser
    {
        /// <summary>
        /// The URL 
        /// </summary>
        string PostUrl { get; }

        /// <summary>
        /// This is the UriPartial.Authority from the server that sent us a response.  
        /// This is important when the PostsUrl is relative.
        /// </summary>
        string UrlAuthority { set; }

        /// <summary>
        /// Load the response received from a server
        /// </summary>
        /// <param name="stream"></param>
        void LoadResponse(Stream stream);

        /// <summary>
        /// Return the post data necessary to take the next step.
        /// </summary>
        /// <returns></returns>
        PostData PostData { get; }
    }
}
